src/app/clusters/joint-fabric-administrator-server/joint-fabric-administrator-server.cpp

```diff
@@ -158,15 +193,73 @@ bool emberAfJointFabricAdministratorClusterTransferAnchorCompleteCallback(
     return true;
 }
 
-// TODO
 bool emberAfJointFabricAdministratorClusterAnnounceJointFabricAdministratorCallback(
     chip::app::CommandHandler * commandObj, const chip::app::ConcreteCommandPath & commandPath,
     const chip::app::Clusters::JointFabricAdministrator::Commands::AnnounceJointFabricAdministrator::DecodableType & commandData)
 {
     MATTER_TRACE_SCOPE("AnnounceJointFabricAdministrator", "JointFabricAdministrator");
 
+    ChipLogProgress(JointFabric, "emberAfJointFabricAdministratorClusterAnnounceJointFabricAdministratorCallback: %u",
+                    commandData.endpointID);
+
+    auto nonDefaultStatus = Status::Success;
+
+    auto & failSafeContext = Server::GetInstance().GetFailSafeContext();
+    VerifyOrExit(failSafeContext.IsFailSafeArmed(commandObj->GetAccessingFabricIndex()),
```

Comment:
I don't see this check in the spec.  Where is this coming from?

Reply:
See spec fix: now both the ICACCSRRequest and AddICAC are required to be received over CASE.

Reply:
Correct, please check the latest version of the PR:

```
// If current fabric is not available, command was invoked over PASE which is not legal
VerifyOrExit(fabricInfo != nullptr, nonDefaultStatus = Status::InvalidCommand);
```

And fabricInfo is extracted using:

```
const FabricInfo * RetrieveCurrentFabric(CommandHandler * aCommandHandler)
{
    FabricIndex index = aCommandHandler->GetAccessingFabricIndex();
    ChipLogDetail(Zcl, "JointFabric: Finding fabric with fabricIndex 0x%x", static_cast<unsigned>(index));
    return Server::GetInstance().GetFabricTable().FindFabricWithIndex(index);
}
```